src/bindings/build.py

```diff
@@ -34,6 +34,17 @@
     with open(header, "r") as hfile:
         ffi.cdef(hfile.read())
 
+# we need to easily control a few values for Windows builds
+source = []
+if os.getenv("PYNACL_SODIUM_STATIC") is not None:
+    source.append("#define SODIUM_STATIC")
+
+source.append("#include <sodium.h>")
+
+if os.getenv("PYNACL_SODIUM_LIBRARY_NAME") is not None:
+    libraries = [os.getenv("PYNACL_SODIUM_LIBRARY_NAME")]
```

Comment:
The Windows library naming scheme makes this somewhat useful. When you ask for "sodium" on linux/OS X it gets translated to "libsodium.so" or "libsodium.dylib" but on windows it just looks for "sodium.lib". The libsodium precompiled libraries, however, call it "libsodium" on windows. We could just rename it to "sodium" but it's nice to be able to use the libsodium.org download without renaming a bunch of files.

Alternately we could just define the library name to be "libsodium" on Windows, but that does mean windows users (at least the ones who want to compile it themselves) need to know that we require the library to be named that.

I'm neutral on which approach we take here (although there is a reasonable argument to be made that it's unlikely anyone on Windows is going to name it anything other than "libsodium")